BlueprintUI/Sources/Internal/LayoutTreeNode.swift

```diff
@@ -0,0 +1,72 @@
+import CoreGraphics
+import Foundation
+
+/// A node in the layout tree.
+///
+/// This tree is created during layout and measurement, and is used to hold information that may be
+/// reused if a node is visited multiple times, such as measurement caches.
+///
+/// In the layout tree, each edge is keyed by an `ElementIdentifier`, and each node corresponds to
+/// an `ElementContent`. An `ElementPath` can uniquely identify a node from the root. Elements that
+/// lazily generate subelements, such as `GeometryReader`, may create subelements with distinct
+/// identifiers across visits during a single render, which will result in spurious nodes being
+/// created that do not correspond to `LayoutResultNode` in the final layout. However, the
+/// `LayoutResultNode` tree will be an isomorphic subgraph of this tree (every `LayoutResultNode`
+/// corresponds to a `LayoutTreeNode`, but not every `LayoutTreeNode` produces a
+/// `LayoutResultNode`).
+///
+final class LayoutTreeNode {
+
+    typealias Subnode = LayoutTreeNode
+    typealias SubnodeKey = ElementIdentifier
+
+    private var subnodes: [SubnodeKey: Subnode] = [:]
+
+    let path: String
+    let sizeCache: HintingSizeCache
+
+    // These commonly used properties are strongly typed. If we need to hang more generalized
+    // things off this type we may want to store them in a heterogeneous dictionary.
+    private var _layoutSubelements: [LayoutSubelement]?
+    private var _associatedCache: Any?
```

Comment:
Sorry, is this strongly typed? Even if the underlying access is in practice, I wonder if the `Any?` box storage incurs a bit of a cost.